src/back/routes/update.ts

```diff
@@ -179,3 +181,89 @@ async function deleteProposalUpdate(req: WithAuth<Request<{ update_id: string }>
 
   return true
 }
+
+function parseFinancialRecords(financial_records: unknown) {
+  const parsedResult = FinancialUpdateSectionSchema.safeParse({ financial_records })
+  if (!parsedResult.success) {
+    ErrorService.report('Submission of invalid financial records', {
+      error: parsedResult.error,
+      category: ErrorCategory.Financial,
+    })
+    throw new RequestError(`Invalid financial records`, RequestError.BadRequest)
+  }
+  return parsedResult.data.financial_records
+}
+
+async function validateFinancialRecords(
+  project: Project,
+  financial_records: unknown
+): Promise<FinancialRecord[] | null> {
+  const { proposal_id, vesting_addresses } = project
+  const [vestingData, updates] = await Promise.all([
+    VestingService.getVestingInfo(vesting_addresses),
+    UpdateService.getAllByProposalId(proposal_id),
```

Comment:
I'd be good to find all usages of this and switch to projects (I don't think we are going to show updates with proposals anymore right?)